packages/noya-state/src/reducers/symbolsReducer.ts

```diff
@@ -0,0 +1,175 @@
+  | [type: 'setLayoutAnchor', value: Sketch.InferredLayoutAnchor]
+  | [type: 'setMinWidth', value: number]
+  | [type: 'setAllowsOverride', value: boolean];
+export function symbolsReducer(
+  state: ApplicationState,
+  action: SymbolsAction,
+): ApplicationState {
+  switch (action[0]) {
+    case 'setAdjustContentOnResize': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.resizesContent = value;
+      });
+    }
+    case 'setHasBackgroundColor': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.hasBackgroundColor = value;
+      });
+    }
+    case 'setBackgroundColor': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.backgroundColor = value;
+      });
+    }
+    case 'setIncludeBackgroundColorInExport': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.includeBackgroundColorInExport = value;
+      });
+    }
+    case 'setIncludeBackgroundColorInInstance': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.includeBackgroundColorInInstance = value;
+      });
+    }
+    case 'setGroupLayout': {
+      const [, value] = action;
+      const filter =
+        getCurrentTab(state) === 'canvas'
+          ? state.selectedObjects
+          : state.selectedSymbolsIds;
+
+      return produce(state, (draft) => {
+        const symbol = getSymbols(draft).filter((symbol) =>
+          filter.includes(symbol.do_objectID),
+        )[0];
+        symbol.groupLayout =
+          value === undefined || Number.isNaN(value)
```

Comment:
We shouldn't pass around `NaN` values. If we're parsing and getting a `NaN`, we should use a different approach, e.g. passing `undefined`. `NaN` can be tricky to work, so I'd treat it more like an error case and handle it as soon as it pops up